homeassistant/components/evohome/__init__.py

```diff
@@ -214,7 +137,7 @@ async def load_auth_tokens(store) -> tuple[dict, dict | None]:
     try:
         await client_v2.login()
     except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
-        _handle_exception(err)
+        _LOGGER.error(_contextualise_v2_exception(err))
```

Comment:
Same as ^^^ – it would be nice if HASS didn't have to be aware of raw `aiohttp` errors (and could instead watch for `evohome-async` errors that have already performed proper contextualization).

Reply:
... but this is a convincing point.

Lemme look at it.

Reply:
First thought - the coordinator makes a distinction between authentication errors (and wrong password) and other errors... How can I manage that if they're all wrapped up in `EvohomeError`?

Reply:
... and `evohomeasync2.AuthenticationError` is more _authorization_ error (i.e. access token), rather than a _authentication_ error.

Reply:
The general strategy I use with my libraries is to have some sort of base error, then have other, more specific errors inherit from that base error. Then, I can always choose to look for the more specific error or fall back to looking for the base error.

Reply:
> it would be nice if HASS didn't have to be aware of raw aiohttp errors (and could instead watch for evohome-async errors that have already performed proper contextualization

With the exception of `AuthenticationError`, the contextualisation (i.e. helpful error message) is _within the context of HA_, for example:

```python
    except ClientResponseError:
        if err.status == HTTPStatus.TOO_MANY_REQUESTS:
            return (
                "You have exceeded the vendor's API rate limit. If this message "
                f"persists, consider increasing the {CONF_SCAN_INTERVAL}"
            )
```
...and (merging v1/v2 data is performed by the integration, not the client):
```python
    except ClientError:
        return (
            "Unable to obtain the latest high-precision temperatures. "
            "Proceeding with low-precision temperatures. "
            "Check your network and the vendor's service status page. "
            f"Message is: {err}"
        )
    except IndexError:
        return (
            "The v2 API's configured location doesn't match "
            "the v1 API's default location (there is more than one location), "
            "so the high-precision feature will be disabled"
        )
```

The others are simply excessively helpful messages for the less IT-literate:
```python
    except ClientConnectionError:
        return (
            "Unable to connect with the vendor's server. "
            "Check your network and the vendor's service status page. "
            f"Message is: {err}"
        )
```
... previous experience with HA has demonstrated that these instructions to check the vendor's website avoids issues being raised in HA.

If you want these removed from HA, I'm OK with that (but I won't be moving them into the client)>. I'd just remove the 'contextualise' functions, trap HA-specific errors (e.g. `CONF_SCAN_INTERVAL`) and re-raise everything else.